components/Feed/Forms/NewPostForm.tsx

```diff
@@ -219,7 +277,18 @@ const NewPostForm: FC<NewPostProps> = ({
         if (isComparePost()) {
             // This is a compare post and it is slightly more complex than the single image post
             // since now we need to upload two images and/or text to the DB and post
-            let mediaObjectList: { type: string; value: string }[] = []
+            //let mediaObjectList: { type: string; value: string }[] = []
+            let leftMediaObject: { text: string, image: string, previewImage: string } = {
+                text: '',
+                image: '',
+                previewImage: '',
+            }
+            let rightMediaObject: { text: string, image: string, previewImage: string } = {
+                text: '',
+                image: '',
+                previewImage: '',
+            }
+            let mediaObjectList: Array<object> = []
             let votesObjMapList: {}[] = [] // TODO: remove and fix likes
```

Comment:
Looks like you forgot about fixing :)

Reply:
Hahaha, the must have missed it during the rebasing. it was very confusing.